greymatter-io/gitops-core
# Fix cue eval failure due to greymatter-cue update

This PR fixes a compatibility bug between gitops-core and greymatter-cue.

The greymatter-cue repo has recently been updated, and somehow one of the fields changed in a backwardly-incompatible way. This line changes the definition `#ClientCertDetails` to require an all-uppercase "URI" instead of the previous "uri":

https://github.com/greymatter-io/greymatter-cue/pull/31/files#diff-77e2b7fc7b3d2a394bffa8cf837d60832cb5162bff97d9270dc2dc49d82ce12eR519

This line of gm/intermediates.cue uses the lowercase version, so it fails to evaluate: https://github.com/greymatter-io/gitops-core/blob/main/gm/intermediates.cue#L337

This problem only appears when `config.spire==true`, because only then is the problematic part of a `#listener` rendered.

Associated Slack monologue: https://greymatter.slack.com/archives/C07TJSLQ0/p1661378864220859

## To Test

- Clone this branch freshly with `--recurse-submodules`
- Make sure your version of the greymatter-cue submodule (`cue.mod/pkg`) is on main of greymatter-cue.
- Set `config.spire` to `true` in `inputs.cue`.
- Run `cue eval -c ./gm/outputs -e mesh_configs` and confirm you get no errors.
